### ISSUE TITLE
[feature req] Tell students about the test runner's environment

### ISSUE THREAD

**Issue by glennj** [exercism/exercism#6096]
# Feature request

Test runners report on version information (or any other info) that can be relayed back to the Results tab of the online editor.

## Rationale

Students can self-diagnose that unexpected errors are occurring due to version mismatches between the test runner and their local environment.

## Proposal

1. The `results.json` output gets a top-level `test-environment` key of type object. The keys and values of this object are arbitrary, whatever the track's maintainers want to report.
    * For example, the bash track might want to use something like:
        ```json
        {
            "bash version": "5.0.17(1)-release",
            "bats version": "Bats 1.1.0"
        }
        ```

2. Somewhere on the Results tab, there is some UI to display the environment information. Not being graphically inclined, I'll not offer any suggestions for how this should be presented. 
![Results tab](https://user-images.githubusercontent.com/122470/140426235-e2a83f6b-d118-40b7-aaa8-4a85698cd8bd.png)

**Comment by glennj** [comment]
Relates to exercism/bash#557

**Comment by SleeplessByte** [comment]
I also love this for TS and JS. Despite babel, this could help. Same for Ruby 🗡️

**Comment by BethanyG** [comment]
Agree.  Would be helpful for Python as well, particularly since we're not upgrading to `3.10` for a bit.

**Comment by iHiD** [comment]
Like this a lot.

**Comment by NobbZ** [comment]
Similarily for Erlang and Elixir language version might be important when using certain features.

**Comment by glennj** [comment]
Anyone have thoughts on how the results would be displayed? I was thinking of just a modal containing a key:value table.

**Comment by BethanyG** [comment]
This is what I get from `pytest`, which is effectively key and value:

![image](https://user-images.githubusercontent.com/5923094/140581847-edd70f42-3b74-4d47-85cd-e9c6e7029a70.png)

so:  yes to ---  

> I was thinking of just a modal containing a key:value table.

😄

**Comment by glennj** [comment]
What are the next steps to get this implemented? The track-specific stuff is, well, track specific. Where would this request sit in the queue for UI improvements & feature requests?
